css/css-color/parsing/color-mixed-num-pct.html

```diff
@@ -0,0 +1,200 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Color 4: color() parsing, mixed number, percent and none</title>
+<link rel="help" href="https://drafts.csswg.org/css-color-4/#color-function">
+<link rel="author" title="Chris Lilley" href="mailto:chris@w3.org">
+<meta name="assert" content="number, percent and none can be freely intermixed in color function">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="test"></div>
+<script>
+    const div = document.querySelector("#test");
+    function testColorFunction(description, rule, expectedValue) {
+        test(function() {
+            div.style.color = "black";
+            div.style.color = rule;
+            assert_equals(getComputedStyle(div).color, expectedValue);
+        }, description);
+    }
+
+    // Opaque sRGB in color()
+    testColorFunction("sRGB all numbers", "color(srgb 1.00 0.50 0.200)", "color(srgb 1 0.5 0.2)");
+    testColorFunction("sRGB all percent", "color(srgb 100% 50% 20%)", "color(srgb 1 0.5 0.2)");
+    testColorFunction("sRGB mixed number and percent", "color(srgb 100% 0.5 20%)", "color(srgb 1 0.5 0.2)");
+    testColorFunction("sRGB mixed number and percent 2", "color(srgb 1.00 50% 0.2)", "color(srgb 1 0.5 0.2)");
+    testColorFunction("sRGB all none", "color(srgb none none none)", "color(srgb 0 0 0)");
```

Comment:
I understood that `none` should serialize to `none` (instead of `0`) in `color()`.

**CSS Color 4 - 15. Serializing <color> Values**: *For syntactic forms which support missing color components, the value `none` (equivalently `NONE`, `nOnE`, etc), shall be serialized in all-lower case as the string "none"*. 

**CSS Color 4 - 4.4. “Missing” Color Components and the none Keyword**: *If a color with a missing component is serialized or otherwise presented directly to an author, then for legacy color syntax it represents that component as a zero value; otherwise, it represents that component as being the `none` keyword*.